options.go

```diff
@@ -13,44 +13,121 @@ type LoggerFunc func(level codersdk.LogLevel, format string, args ...interface{}
 
 // Options contains the configuration for the envbuilder.
 type Options struct {
-	SetupScript          string
-	InitScript           string
-	InitCommand          string
-	InitArgs             string
-	CacheRepo            string
-	BaseImageCacheDir    string
-	LayerCacheDir        string
-	DevcontainerDir      string
+	// SetupScript is the script to run before the init script. It runs as the
+	// root user regardless of the user specified in the devcontainer.json file.
+	// SetupScript is ran as the root user prior to the init script. It is used to
+	// configure envbuilder dynamically during the runtime. e.g. specifying
+	// whether to start systemd or tiny init for PID 1.
+	SetupScript string
+	// InitScript is the script to run to initialize the workspace.
+	InitScript string
+	// InitCommand is the command to run to initialize the workspace.
+	InitCommand string
+	// InitArgs are the arguments to pass to the init command. They are split
+	// according to /bin/sh rules with https://github.com/kballard/go-shellquote.
+	InitArgs string
+	// CacheRepo is the name of the container registry to push the cache image to.
+	// If this is empty, the cache will not be pushed.
+	CacheRepo string
+	// BaseImageCacheDir is the path to a directory where the base image can be
+	// found. This should be a read-only directory solely mounted for the purpose
+	// of caching the base image.
+	BaseImageCacheDir string
+	// LayerCacheDir is the path to a directory where built layers will be stored.
+	// This spawns an in-memory registry to serve the layers from.
+	LayerCacheDir string
+	// DevcontainerDir is the path to the folder containing the devcontainer.json
+	// file that will be used to build the workspace and can either be an absolute
+	// path or a path relative to the workspace folder. If not provided, defaults
+	// to `.devcontainer`.
+	DevcontainerDir string
+	// DevcontainerJSONPath is the path to a devcontainer.json file that is either
+	// an absolute path or a path relative to DevcontainerDir. This can be used in
+	// cases where one wants to substitute an edited devcontainer.json file for
+	// the one that exists in the repo.
```

Comment:
Some of these comments look quite different from before the serpent refactor.

```suggestion
	// DevcontainerJSONPath is a path to a devcontainer.json file
	// that is either an absolute path or a path relative to
	// DevcontainerDir. This can be used in cases where one wants
	// to substitute an edited devcontainer.json file for the one
	// that exists in the repo.
	// If neither `DevcontainerDir` nor `DevcontainerJSONPath` is provided,
	// envbuilder will browse following directories to locate it:
	// 1. `.devcontainer/devcontainer.json`
	// 2. `.devcontainer.json`
	// 3. `.devcontainer/<folder>/devcontainer.json`
```

Do you remember why "If neither"-section and the list of files was removed?

There was some other changes too, but this was a big one that stood out to me when I compared. IMO there's some valuable information there that was lost.

Reply:
I'm not sure... I may have missed them during copy + paste but I will do a post review to see what we missed.